src/NUnitEngine/nunit.engine/Runners/MasterTestRunner.cs

```diff
@@ -429,9 +435,20 @@ private TestEngineResult RunTests(ITestEventListener? listener, TestFilter filte
 
             _eventDispatcher.OnTestEvent(startRunNode.OuterXml);
 
+            if (_testRunTimeout > 0)
+            {
+                _testRunTimer = new Timer(_testRunTimeout);
+                _testRunTimer.Elapsed += OnTestRunTimeout;
+                _testRunTimer.AutoReset = false;
+                _testRunTimer.Enabled = true;
+            }
+
             long startTicks = Stopwatch.GetTimestamp();
 
-            TestEngineResult result = PrepareResult(GetEngineRunner().Run(_eventDispatcher, filter)).MakeTestRunResult(TestPackage);
+            TestEngineResult result = GetEngineRunner().Run(_eventDispatcher, filter);
+            result = result.MakeTestRunResult(TestPackage);
+            result = PrepareResult(result);
```

Comment:
This looks like a change in the logic.  The previous code is

1. Call `GetEngineRunner().Run(...)`
2. Call `PrepareResult( ... )` on it.
3. Call `MakeTestRunResult(TestPackage)` on it.

In the new code steps 2 and 3 are swapped.  No idea if this has an impact.